### ISSUE TITLE
⌨ `DropdownMenu` component colour contrast insufficient

### ISSUE THREAD

**Issue by mnigh** [GCTC-NTGC/gc-digital-talent#7351]
## ⌨ Accessibility Issue

The `DropdownMenu` component colour contrast insufficient.

### WCAG 2.1 Criteria

SC 1.4.3: Contrast (Minimum) (Level AA). REF: https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html

## 🦋 Expected Behaviour

Text should have sufficient contrast.

## 📋 Steps to Reproduce

1. Navigate to `/search`
2. Click All skills button
3. Observe All skills text in dropdown 

## 📸 Screenshot

![Screen Shot 2023-07-20 at 16 20 20](https://github.com/GCTC-NTGC/gc-digital-talent/assets/3046459/39c49730-749a-4a96-b4a6-448a944b4b9f)

## 🙋‍♀️ Proposed Solution

1. Change item colour from `secondary.dark` to `black`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d244993542fbe7aaef8d9d38ff6212345e02e8e7/packages/ui/src/components/DropdownMenu/DropdownMenu.tsx#L91

2. Change label colour from `gray` to `black`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d244993542fbe7aaef8d9d38ff6212345e02e8e7/packages/ui/src/components/DropdownMenu/DropdownMenu.tsx#L137

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] items appear in a darker colour in the `DropdownMenu` component
- [ ] update `DropdownMenu` story to be open by default to catch contrast issues in the future
